### PR TITLE
feat: add indents.scm for nickel

### PR DESCRIPTION
**Pull request by zhengpd**
This add basic indentation support for nickel files.

### CODE REVIEW

File: queries/nickel/indents.scm [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[
+  (term)
+] @indent.begin
+
+[
+  "]"
+  "}"
```
Comment (lucario387) [inline queries/nickel/indents.scm:7]: This should also be captured with `@indent.end`

File: queries/nickel/indents.scm [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[
+  (term)
```
Comment (lucario387) [inline queries/nickel/indents.scm:2]: Does this mean `(term)` consists of both `[...]` and `{...}` sections

File: tests/indent/nickel_spec.lua [hunk 3]
```diff
@@ -0,0 +1,18 @@
+local Runner = require("tests.indent.common").Runner
+
+local run = Runner:new(it, "tests/indent/nickel", {
+  shiftwidth = 2,
+  expandtab = true,
+})
+
+describe("indent Nickel:", function()
```
Comment (lucario387) [inline tests/indent/nickel_spec.lua:8]: There should be a whole file test here, to check whether it's running or not

File: queries/nickel/indents.scm (hunk 2 above)
Comment (zhengpd) [reply queries/nickel/indents.scm:2]: > Does this mean `(term)` consists of both `[...]` and `{...}` sections

@lucario387 `(term)` is the container of all value types,  including `[...]` and `{...}`

**Review by lucario387 — APPROVED** [review]

**Comment by amaanq** [comment]
squash as `feat(nickel): add indents` @lucario387

**Pull request merged**